apps/examples/websocket/websocket_main.c

```diff
@@ -747,9 +749,17 @@ int websocket_main(int argc, char *argv[])
 			goto error_with_input;
 		}
 
-		strncpy(input->server_ip, argv[2], 20);
-		strncpy(input->server_port, argv[3], 8);
-		strncpy(input->path, argv[4], 32);
+		int addr_len = strlen(argv[2]);
+		int port_len = strlen(argv[3]);
+		int path_len = strlen(argv[4]);
+
+		if (addr_len > WEBSOCKET_EXAMPLE_ADDR_LEN || port_len > WEBSOCKET_EXAMPLE_PORT_LEN
```

Comment:
you can use below.
#define WEBSOCKET_EXAMPLE_ADDR_LEN 8
if (addr_len >= WEBSOCKET_EXAMPLE_ADDR_LEN) {
      goto error;
}
strncpy(input->server_ip, argv[2], WEBSOCKET_EXAMPLE_ADDR_LEN);

Reply:
I alloced buffer WEBSOCKET_EXAMPLE_ADDR_LEN + 1 so it is safe

Reply:
I mean +1 is not needed

Reply:
+1 is needed for null termination.
if (addr_len > WEBSOCKET_EXAMPLE_ADDR_LEN) looks good to me.

Reply:
I checked now, my comment was if you want avoid enter string which size is 8, 20, 32,
changing addr_len > X to addr_len >= X
That was my comment.
I just post this message as a reference.